app/main.cpp

```diff
@@ -214,6 +214,12 @@ int main(int argc, char *argv[])
                                    QLatin1String("max-frames"), QLatin1String("64"));
     parser.addOption(maxFrames);
 
+    QCommandLineOption maxStack(QLatin1String("max-stack"),
```

Comment:
how is this different from max-frames? can't this whole patch be dropped now?

Reply:
It can not be dropped at least because there is no any constraints on max-frame value in resolveCallchain.
And max-stack is compatible with 'perf' max-stack option. 
'perf' has the similar constraint on max-stack value in its code when it resolves calchains.

Do you want me to exchange max-stack with max-frame?
I don't see reasons to mix these options.

Reply:
From my POV it is not clear what the difference between max-stack and max-frame should be. Both define the maximum depth of the unwound stack, no?

Our command line options aren't compatible with perf anyways, so that doesn't matter.

I would say: drop the max-stack option, and then add the check for max-frames where you think it's missing.

Reply:
What about related to 'max-stack' option hotspot's part and lazy stack unwinding? Should I also drop that stuff?

Reply:
that sounds like it's totally unrelated to the disassembler. please let us concentrate on that one first and talk about this other feature of yours once his work here has landed.

as I mentioned at the start - I haven't even looked at the hotspot related change set yet, as I first want to ensure the perfparser changes are sane

Reply:
Ok